docs/src/adr/023adr-trace-evaluation.md

```diff
@@ -0,0 +1,411 @@
+                    "kind": "ValEx",
+                    "value": {
+                      "kind": "TlaInt",
+                      "value": 4
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+        },
+        {
+          "type": "() => Bool",
+          "kind": "TlaOperDecl",
+          "name": "State4",
+          "formalParams": [
+            
+          ],
+          "isRecursive": false,
+          "body": {
+            "type": "Bool",
+            "kind": "OperEx",
+            "oper": "AND",
+            "args": [
+              {
+                "type": "Bool",
+                "kind": "OperEx",
+                "oper": "EQ",
+                "args": [
+                  {
+                    "type": "Int",
+                    "kind": "NameEx",
+                    "name": "x"
+                  },
+                  {
+                    "type": "Int",
+                    "kind": "ValEx",
+                    "value": {
+                      "kind": "TlaInt",
+                      "value": 5
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+        },
+        {
+          "type": "() => Bool",
+          "kind": "TlaOperDecl",
+          "name": "InvariantViolation",
+          "formalParams": [
+            
+          ],
+          "isRecursive": false,
+          "body": {
+            "type": "Bool",
+            "kind": "ValEx",
+            "value": {
+              "kind": "TlaBool",
+              "value": true
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+</details>
+
+After running `tracee --trace=testTrace.json --expressions=A,B,C,D test.tla`, we should see
+```
+...
+Constructing an example run                                       I@16:06:59.450
+Check the trace in: <PATH>/example0.tla, ... I@16:06:59.563
+The outcome is: NoError                                           I@16:06:59.571
+Trace successfully generated.   
+```
+
+where `example0.tla` looks like
+
+<details>
+<summary>Result</summary>
+
+```tla
+---------------------------- MODULE counterexample ----------------------------
+
+EXTENDS test
+
+(* Constant initialization state *)
+ConstInit == TRUE
+
+(* Initial state *)
+State0 == A = 1/\ B = 0/\ C = {1}/\ D = FALSE
+
+(* Transition 0 to State1 *)
+State1 == A = 4/\ B = 0/\ C = {2}/\ D = TRUE
+
+(* Transition 1 to State2 *)
+State2 == A = 9/\ B = 1/\ C = {}/\ D = FALSE
+
+(* Transition 2 to State3 *)
+State3 == A = 16/\ B = 1/\ C = {4}/\ D = TRUE
+
+(* Transition 3 to State4 *)
+State4 == A = 25/\ B = 1/\ C = {}/\ D = FALSE
+
+(* The following formula holds true in the last state and violates the invariant *)
+InvariantViolation == TRUE
+
+================================================================================
+(* Created by Apalache on Mon Oct 17 16:06:59 CEST 2022 *)
+(* https://github.com/informalsystems/apalache *)
+```
```

Comment:
Thanks! do find having the example here really helps.

After our discussion today, I think it would be worth including a note about how operators can be easily added to get views into regions of the state one may be interested in.

Also, I do think it's worth considering the interaction with the trace viewer, and I do think @thpani makes a strong case here.

IMO, it's fine if we decide to not pursue the course he advises (to extend the state with these  evaluations, and leave filtering to the viewer), but the rationale for that choice is probably good to record here (perhaps it's just because it takes less work! :) .

With those additions, for my part, I'm quite happy to have this merged in.

Reply:
That's a good point, maybe we can include a link to https://github.com/informalsystems/vscode-itf-trace-viewer and note that it supports filtering by variable?